Source/Photos.swift

```diff
@@ -0,0 +1,62 @@
+import Photos
+import AssetsLibrary
+
+struct Photos {
+
+  static func fetch(completion: (assets: [PHAsset]) -> Void) {
+    let fetchOptions = PHFetchOptions()
+    let authorizationStatus = ALAssetsLibrary.authorizationStatus()
+    var fetchResult: PHFetchResult?
+
+    guard authorizationStatus == .Authorized else { return }
+
+    if fetchResult == nil {
+      fetchResult = PHAsset.fetchAssetsWithMediaType(PHAssetMediaType.Image, options: fetchOptions)
+    }
+
+    if fetchResult?.count > 0 {
+      var assets = [PHAsset]()
+      fetchResult?.enumerateObjectsUsingBlock { object, index, stop in
+        if let asset = object as? PHAsset {
+          assets.insert(asset, atIndex: 0)
+        }
+      }
+
+      dispatch_async(dispatch_get_main_queue(), {
+        completion(assets: assets)
+      })
+    }
+  }
+
+  static func resolveAsset(asset: PHAsset, size: CGSize = CGSize(width: 720, height: 1280) ,completion: (image: UIImage?) -> Void) {
+    let imageManager = PHImageManager.defaultManager()
+    let requestOptions = PHImageRequestOptions()
+
+    imageManager.requestImageForAsset(asset, targetSize: size, contentMode: PHImageContentMode.AspectFill, options: requestOptions, resultHandler: { image, info in
+      if let info = info where info["PHImageFileUTIKey"] == nil {
```

Comment:
This is a bit dirty but it helps it from not calling the completion twice. This has to do with how `requestImageForAsset` resolves assets.

> For an asynchronous request, Photos may call your result handler block more than once. Photos first calls the block to provide a low-quality image suitable for displaying temporarily while it prepares a high-quality image. (If low-quality image data is immediately available, the first call may occur before the method returns.)

http://stackoverflow.com/questions/27670325/photos-framework-requestimageforasset-returning-two-results-cant-set-image-vi

@hyperoslo/mystore-ios